builtin/providers/azurerm/resource_arm_image_test.go

```diff
@@ -0,0 +1,343 @@
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for image: %s", dName)
+		}
+
+		conn := testAccProvider.Meta().(*ArmClient).imageClient
+
+		resp, err := conn.Get(resourceGroup, dName, "")
+		if err != nil {
+			return fmt.Errorf("Bad: Get on imageClient: %s", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound && shouldExist {
+			return fmt.Errorf("Bad: Image %q (resource group %q) does not exist", dName, resourceGroup)
+		}
+		if resp.StatusCode != http.StatusNotFound && !shouldExist {
+			return fmt.Errorf("Bad: Image %q (resource group %q) still exists", dName, resourceGroup)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureVMExists(sourceVM string, shouldExist bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		log.Printf("[INFO] testing MANAGED IMAGE VM EXISTS - BEGIN.")
+
+		vmClient := testAccProvider.Meta().(*ArmClient).vmClient
+		vmRs, vmOk := s.RootModule().Resources[sourceVM]
+		if !vmOk {
+			return fmt.Errorf("VM Not found: %s", sourceVM)
+		}
+		vmName := vmRs.Primary.Attributes["name"]
+
+		resourceGroup, hasResourceGroup := vmRs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for VM: %s", vmName)
+		}
+
+		resp, err := vmClient.Get(resourceGroup, vmName, "")
+		if err != nil {
+			return fmt.Errorf("Bad: Get on vmClient: %s", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound && shouldExist {
+			return fmt.Errorf("Bad: VM %q (resource group %q) does not exist", vmName, resourceGroup)
+		}
+		if resp.StatusCode != http.StatusNotFound && !shouldExist {
+			return fmt.Errorf("Bad: VM %q (resource group %q) still exists", vmName, resourceGroup)
+		}
+
+		log.Printf("[INFO] testing MANAGED IMAGE VM EXISTS - END.")
+
+		return nil
+	}
+}
+
+func testCheckAzureRMImageDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*ArmClient).diskClient
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_managed_disk" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := conn.Get(resourceGroup, name)
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Managed Disk still exists: \n%#v", resp.Properties)
+		}
+	}
+
+	return nil
+}
+
+var testAccAzureRMImage_standaloneImage = `
+resource "azurerm_resource_group" "test" {
+    name = "acctestRG-%[1]d"
+    location = "West Central US"
+}
+
+resource "azurerm_image" "test" {
+    name = "accteste-%[1]d"
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+	os_disk_os_type = "Linux"
+	os_disk_os_state = "Generalized"
+	os_disk_blob_uri = "https://terraformdev.blob.core.windows.net/packerimages/ubuntu_plain.vhd"
```

Comment:
> Note that two acceptance tests currently require subscription-specific information (VHD has to be in a user account or VM must be generalized). This information would need to be provided by the person running the tests - open to ideas on improving this.

We could use a multi-part test here - spinning up a VM first, generalising it, then using the image in third test-case - which would allow this to work in any subscription?